packages/graphql-server-express/src/expressApollo.ts

```diff
@@ -42,33 +42,48 @@ export function graphqlExpress(options: GraphQLOptions | ExpressGraphQLOptionsFu
     }
 
     const formatErrorFn = optionsObject.formatError || graphql.formatError;
+    let requestPayload;
+
+    switch ( req.method ) {
+      case 'POST':
+        if ( !req.body ) {
+          res.statusCode = 500;
+          res.write('POST body missing. Did you forget "app.use(bodyParser.json())"?');
+          res.end();
+          return;
+        }
 
-    if (req.method !== 'POST') {
-      res.setHeader('Allow', 'POST');
-      res.statusCode = 405;
-      res.write('Apollo Server supports only POST requests.');
-      res.end();
-      return;
-    }
+        requestPayload = req.body;
+        break;
+      case 'GET':
+        if ( !req.query ) {
+          res.statusCode = 500;
```

Comment:
@DxCx right, but you're checking req.query. If that's not present, it means it wasn't supplied by the user. In that case I would expect a 400, not a 500.